druid-handler/src/java/org/apache/hadoop/hive/druid/DruidStorageHandler.java

```diff
@@ -98,12 +306,89 @@ public void rollbackDropTable(Table table) throws MetaException {
 
   @Override
   public void commitDropTable(Table table, boolean deleteData) throws MetaException {
-    // Nothing to do
+    if (MetaStoreUtils.isExternalTable(table)) {
+      return;
+    }
+    String dataSourceName = Preconditions
+            .checkNotNull(table.getParameters().get(Constants.DRUID_DATA_SOURCE),
+                    "DataSource name is null !"
+            );
+
+    if (deleteData == true) {
+      LOG.info(String.format("Dropping with purge all the data for data source [%s]",
+              dataSourceName
+      ));
+      List<DataSegment> dataSegmentList = DruidStorageHandlerUtils
+              .getDataSegmentList(connector, druidMetadataStorageTablesConfig, dataSourceName);
+      if (dataSegmentList.isEmpty()) {
+        LOG.info(String.format("Nothing to delete for data source [%s]", dataSourceName));
+        return;
+      }
+      for (DataSegment dataSegment : dataSegmentList) {
+        try {
+          deleteSegment(dataSegment);
+        } catch (SegmentLoadingException e) {
+          LOG.error(String.format("Error while deleting segment [%s]", dataSegment.getIdentifier()),
+                  e
+          );
+        }
+      }
+    }
+    if (DruidStorageHandlerUtils
+            .disableDataSource(connector, druidMetadataStorageTablesConfig, dataSourceName)) {
```

Comment:
if deleteData is true, should we remove segment entries from metadata storage instead of disabling them ?